playwright/all_tests/assignment/assignmentVersionHistory.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { ContentType, loginTestGroup, Role } from '../../helpers'
+import { test } from '@playwright/test'
+import { Exam } from 'web/src/types'
+import { AssignmentContentModel } from '../../models/AssignmentContentModel'
+import { VersionHistoryModel } from '../../models/VersionHistoryModel'
+import { FormModel } from '../../models/FormModel'
+import { AssignmentFormModel } from '../../models/AssignmentFormModel'
+
+loginTestGroup(test, Role.YLLAPITAJA)
+Object.values(Exam).forEach((exam) => {
+  test.describe(`${exam} assignment version history`, () => {
+    test.beforeEach(async ({ page }) => await new FormModel(page, exam).showKeys())
+
+    test('can create multiple versions of assignment, browse versions and recover', async ({
+      page,
+      context,
+      baseURL
+    }) => {
+      const form = new AssignmentFormModel(page, exam)
+      const content = new AssignmentContentModel(page, exam)
+      const versionHistory = new VersionHistoryModel(page, exam, ContentType.koetehtavat)
+
+      const apiCalls = form.assignmentApiCalls(context, baseURL!)
+
+      const assignmentIn = form.testAssignmentIn('Versiotesti')
+      const assignment = await apiCalls.create(assignmentIn)
+      // Luo 4 versiota
+      const arr = [...Array(4)]
+      for (const [index] of arr.entries()) {
+        await apiCalls.update(
+          assignment.id,
+          JSON.stringify({ ...assignment, nameFi: `${assignment.nameFi} v${index + 1}` })
+        )
+      }
```

Comment:
Tää `arr` on vähän jännä, ku se on vaan 4:n pituinen array undefined-arvoja. O_o

Pitäisköhän ton for-of:in sijaan käyttää perinteistä for-looppia:

```suggestion
      const versionCount = 4
      for (let version = 1; version <= 4; version++) {
        await apiCalls.update(
          assignment.id,
          JSON.stringify({ ...assignment, nameFi: `${assignment.nameFi} v${version}` })
        )
      }
```

Reply:
array pitäs passata viel assertVersionHistoryModal() funkkarille. toki voi ton versionCountin sinne passata ja siel myös tehdä for looppi